cpp/ql/src/semmle/code/cpp/ir/implementation/unaliased_ssa/internal/SSAConstruction.qll

```diff
@@ -26,13 +26,9 @@ cached private module Cached {
       hasChiNode(_, oldInstruction)
     } or
     UnreachedTag(OldInstruction oldInstruction, EdgeKind kind) {
-      // We need an `Unreached` instruction for the destination of any edge whose predecessor
-      // instruction is reachable, but whose successor block is not. This should occur only for
-      // infeasible edges.
-      exists(OldIR::Instruction succInstruction |
-        succInstruction = oldInstruction.getSuccessor(kind) and
-        not succInstruction instanceof OldInstruction
-      )
+      // We need an `Unreached` instruction for the destination of each infeasible edge whose
+      // predecessor is reachable.
+      Reachability::isInfeasibleInstructionSuccessor(oldInstruction, kind)
```

Comment:
Why not have just a single `Unreached` instruction, shared by the whole function?

Reply:
I considered that, but eventually went with one `Unreached` per edge to avoid introducing "critical" edges (edges whose predecessor block has multiple successors and whose successor block has multiple predecessors). Critical edges make it difficult to insert inference operators. For these edges, though, I guess we wouldn't have anything interesting to infer anyway, so I'll try folding all of the `Unreached` instructions into one per function.